File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzerror.c
const char * ZEXPORT gzerror(gzFile file, int *errnum) {
    gz_statep state;

    /* get internal structure and check integrity */
    if (file == NULL)
        return NULL;
    state = (gz_statep)file;
    if (state->mode != GZ_READ && state->mode != GZ_WRITE)
        return NULL;

    /* return error information */
    if (errnum != NULL)
        *errnum = state->err;
    return state->err == Z_MEM_ERROR ? "out of memory" :
                                       (state->msg == NULL ? "" : state->msg);
}

File: src/python/inputs-complex/zlib-1.3.1/individual-funcs/gzerror.rs
fn gzerror(file: &gzFile, errnum: &mut i32) -> Option<&str> {
    let mut state = unsafe { &*(file as *const _ as *mut _) };
    if state.mode != GZ_READ && state.mode != GZ_WRITE {
        return None;
    }
    if errnum != null {
        *errnum = state.err;
    }
    Some(state.msg)
}